cmd/cluster/transferowner.go

```diff
@@ -54,27 +68,247 @@ func newTransferOwnerOptions(streams genericclioptions.IOStreams, globalOpts *gl
+		}
+		fmt.Printf("Pod %s in namespace %s has been deleted.\n", pod.Name, namespace)
+	}
+
+	fmt.Printf("Pods in namespace %s with label selector '%s' have been deleted.\n", namespace, selector)
+	return nil
+}
+
 func (o *transferOwnerOptions) run() error {
-	fmt.Print("Before making changes in OCM, the cluster must have the pull secret updated to be the new owner's. ")
-	fmt.Print("See: https://github.com/openshift/ops-sop/blob/master/v4/howto/replace-pull-secret.md\n")
+	fmt.Println("Notify the customer before ownership transfer commences. Sending service log.")
+	postCmd := generateServiceLog(o.clusterID, "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/maintenance_starting.json")
+	if err := postCmd.Run(); err != nil {
+		fmt.Println("Failed to generate service log. Please manually send a service log to Notify the customer before ownership transfer commences:")
+		fmt.Printf("osdctl servicelog post %v -t %v -p %v\n",
+			o.clusterID, "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/maintenance_starting.json", strings.Join(postCmd.TemplateParams, " -p "))
+	}
 
 	// Create an OCM client to talk to the cluster API
 	// the user has to be logged in (e.g. 'ocm login')
 	ocm, err := utils.CreateConnection()
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create OCM client: %w", err)
 	}
 	defer func() {
 		if ocmCloseErr := ocm.Close(); ocmCloseErr != nil {
 			fmt.Printf("Cannot close the ocm (possible memory leak): %q", ocmCloseErr)
 		}
 	}()
+	cluster, err := utils.GetClusterAnyStatus(ocm, o.clusterID)
+	o.cluster = cluster
+	o.clusterID = cluster.ID()
+
+	userDetails, err := ocm.AccountsMgmt().V1().Accounts().Account(o.newOwnerName).Get().Send()
+	userName, ok := userDetails.Body().GetUsername()
+	if !ok {
+		return fmt.Errorf("Failed to get username from new user id")
+	}
 
 	// Find and setup all resources that are needed
+	hiveCluster, err := utils.GetHiveCluster(o.clusterID)
+	hiveKubeCli, _, hivecClientset, err := getKubeConfigAndClient(hiveCluster.ID())
+	if err != nil {
+		return fmt.Errorf("failed to retrieve Kubernetes configuration and client for Hive cluster ID %s: %w", hiveCluster.ID(), err)
+	}
 
-	cluster, err := utils.GetCluster(ocm, o.clusterID)
+	response, err := ocm.AccountsMgmt().V1().AccessToken().Post().Impersonate(userName).Parameter("body", nil).Send()
 	if err != nil {
-		return err
+		return fmt.Errorf("Can't send request: %w", err)
+	}
+
+	auths, ok := response.Body().GetAuths()
+	if !ok {
+		return fmt.Errorf("Error validating pull secret structure. This shouldn't happen, so you might need to contact SDB")
+	}
+	authsMap := map[string]map[string]string{}
+	for k, auth := range auths {
+		authsMap[k] = map[string]string{
+			"auth":  auth.Auth(),
+			"email": auth.Email(),
+		}
+	}
+
+	pullSecret, err := json.Marshal(map[string]map[string]map[string]string{
```

Comment:
Where's this pull-secret from? In the[ SOP](https://github.com/openshift/ops-sop/blob/master/v4/howto/replace-pull-secret.md#verify-the-new-secret), this pull-secret is from the output of this command
`ocm post --body=/dev/null --header="Impersonate-User=${NEW_OWNER_USERNAME}" /api/accounts_mgmt/v1/access_token > pull-secret.json
`

Reply:
@lnguyen1401 [line 270](https://github.com/reedcort/osdctl/blob/OSD-17502/cmd/cluster/transferowner.go#L270-L274) is where the pull secret is taken from.

Reply:
so is the output from 270 the same as the output from the ocm command above ?

Reply:
yes